lib/cloud/registry.js

```diff
@@ -72,6 +74,16 @@ export class Registry {
     this.funcMap.op[name] = func;
   }
 
+  registerEvent(name, func, options) {
+    const opts = {
+      ...options,
+      name
+    };
+
+    this._addParam('event', opts);
+    this.funcMap.event[name] = func;
```

Comment:
We will need to support multiple func hook on to one event. So the name cannot just set according to the event name.